db/seed.js

```diff
@@ -1,39 +1,37 @@
 require('dotenv').config();
-const faker = require('faker');
 const { MongoClient } = require('mongodb');
+const cluster = require('cluster');
+const numCPUs = require('os').cpus().length;
+const generateDocument = require('./generateDocument');
 
 const dbAddress = process.env.DB_ADDRESS || 'localhost';
 const dbName = process.env.DB_NAME || 'marzagat_overview';
 const collectionName = process.env.COLLECTION_NAME || 'restaurants';
 const seedNum = parseInt(process.env.SEED_NUM, 10) || 10000000;
-const batchSize = parseInt(process.env.SEED_BATCH_SIZE, 10) || 5000;
+const batchSize = parseInt(process.env.SEED_BATCH_SIZE, 10) || 15000;
+const schemaIdName = process.env.SCHEMA_ID_NAME || '_id';
 
-const url = `mongodb://${dbAddress}:27017`;
+const connectToDb = async () => {
+  const url = `mongodb://${dbAddress}/`;
+  const client = await MongoClient.connect(url);
+  const collection = client.db(dbName).collection(collectionName);
+  return { client, collection };
+};
 
-const generateDocument = id => (
-  {
-    _id: id,
-    name: faker.commerce.productName(),
-    tagline: faker.lorem.words((id % 4) + 3),
-    type: faker.lorem.word(),
-    vicinity: `${faker.address.streetAddress()}, ${faker.address.city()}`,
-    priceLevel: faker.random.number({ min: 1, max: 4 }),
-    zagatFood: faker.finance.amount(0, 5, 1),
-    zagatDecor: faker.finance.amount(0, 5, 1),
-    zagatService: faker.finance.amount(0, 5, 1),
-    longDescription: faker.lorem.paragraph((id % 2) + 11),
-  }
-);
+const indexDb = async (collection) => {
+  const indexObj = {};
+  indexObj[schemaIdName] = 1;
+  await collection.ensureIndex(indexObj, { unique: true });
```

Comment:
wanted to confirm what you are doing here. seems like ensureIndex is going through and checking if an index for your schemaIdName already exists and if not create one? does your db need this in order to have indexes or is it just an extra fallback for best practice? i briefly looked into the docs and it seems like ensureIndex might be deprecated, did you consider createIndex instead? https://docs.mongodb.com/manual/reference/method/db.collection.ensureIndex/

Reply:
yeah, thanks I didn't see that! I'll update to createIndex.